tests/Storage/FileSystemStorageTest.php

```diff
@@ -341,24 +342,63 @@ public function testReplacingFileIsCorrectlyUploaded(): void
         $this->storage->upload($this->object, $this->mapping);
     }
 
+    /**
+     * @group upload
+     */
+    public function testReplacingFileWithDirectoryNamerIsCorrectlyUploaded(): void
+    {
+        $file = $this->getReplacingFileMock();
+        $file
+            ->method('getClientOriginalName')
+            ->willReturn('test.txt');
+        $file
+            ->method('getPathname')
+            ->willReturn($this->getValidUploadDir().'/test.txt');
+
+        $this->mapping
+            ->expects(self::once())
+            ->method('getFile')
+            ->with($this->object)
+            ->willReturn($file);
+
+        $this->mapping
+            ->expects(self::once())
+            ->method('getUploadDestination')
+            ->willReturn($this->root->url().\DIRECTORY_SEPARATOR.'storage');
+
+        $this->mapping
+            ->expects(self::once())
+            ->method('getUploadName')
+            ->with($this->object)
+            ->willReturn('test.txt');
+
+        $this->mapping
+            ->expects(self::once())
+            ->method('getUploadDir')
+            ->with($this->object)
+            ->willReturn('vich_uploader_bundle/directoryNamer/1/');
+
+        $this->storage->upload($this->object, $this->mapping);
+    }
+
     public function filenameWithDirectoriesDataProvider(): array
     {
         return [
             // upload dir, dir, expected dir
             [
-                '/root_dir',
+                '/storage/vich_uploader_bundle',
```

Comment:
can we provide useful keys, here instead of relying on numeric keys? E.g. something like "directory with subdirectory" and so on

Reply:
No, this is a [`DataProvider`](https://phpunit.de/manual/3.7/en/writing-tests-for-phpunit.html#writing-tests-for-phpunit.data-providers) for the [unit test above](https://github.com/dustin10/VichUploaderBundle/blob/1d891f943b2cd6fe15b5befe572cd5b78a2fcf4f/tests/Storage/FileSystemStorageTest.php#L262), and I do not see anything in the documentation that this functionality supports named parameters.

Reply:
I mean names for data, not for parameters. See https://phpunit.readthedocs.io/en/9.5/writing-tests-for-phpunit.html?#writing-tests-for-phpunit-data-providers-examples-datatest1-php

Reply:
changed

Reply:
Thanks. What about omitting "upload with "? It looks like labels like "two subdirectories" are enough to understand the case

Reply:
Yes, changed again